feat(authors): introduce reusable AuthorCard component

gativarshney opened the pull request:
# Context

This PR addresses **[Issue #2: feat(authors): introduce reusable AuthorCard component](https://github.com/rudra-iitm/openprinting.github.io/issues/2)**.

The goal is to introduce a reusable, consistent `AuthorCard` UI component that can be used across different pages in the project. This centralizes author presentation and removes duplicated UI across layouts.

---

# Description

This work includes creating a fully reusable `AuthorCard` component that matches the intended design and behavior for the OpenPrinting site. The component includes proper spacing, alignment, fallback handling, and a consistent structure for displaying author information.

It is prepared for use in blog posts, news posts, and any future sections that require author details.

---

# Changes in the Codebase

- Added a new `AuthorCard` component with:
  - Correct layout and alignment matching the project’s design
  - Fallbacks for missing images or incomplete author fields
  - Clean display of name, role, location, email, and GitHub

- Updated `authors.ts` with structured, clean author metadata

- Added author images under `public/authors/`

- Ensured the component is self-contained and ready for reuse across multiple layouts

---

# Additional Information

- Only the relevant files for the AuthorCard component were added or updated
- UI was tested locally to verify layout, spacing, and fallback behavior

rudra-iitm commented on components/AuthorCard.tsx:
```diff
@@ -0,0 +1,115 @@
+}
+
+export default function AuthorCard({ name, className }: Props) {
+  const author: Author | undefined = authors.find(a => a.name === name);
```
Instead of matching the full author name passed through the prop, match it with the names used in the current MD files — for example, "Till" for Till Kamppeter, "Mike" for Michael Sweet, and so on.

You can follow this for ref - https://github.com/OpenPrinting/openprinting.github.io/blob/60e8017f37dee86b9486cd612c3941eb4716acf1/_data/authors.yml

rudra-iitm commented on data/authors.ts:
```diff
@@ -0,0 +1,59 @@
+  image?: string;
+}
+
+const authors: Author[] = [
```
Accordingly update this too.

gativarshney commented:
Hi Rudra, 
I've implemented all the requested changes:

- AuthorCard now accepts **authorKey** instead of the full author object.
- The lookup now matches the short keys used in the MD files (Till, Mike, Aveek, etc.).
- Updated data/authors.ts to include these keys.
- UI tested again locally to confirm everything works as expected.

Please let me know if anything else needs to be refined!

rudra-iitm commented:
This looks nice, merging it 😊. Thanks for the PR.

rudra-iitm commented:
Hi @gativarshney, I got some linting issues after merging your PR. Could you please fix them in a new PR? I’ve reverted the changes for now.
